linux/just_files/just_archive.bsh

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env false bash
+
+if [[ ${-} != *i* ]]; then
+  source_once &> /dev/null && return 0
+fi
+
+# dependencies
+source "${VSI_COMMON_DIR}/linux/aliases.bsh"
+source "${VSI_COMMON_DIR}/linux/command_tools.bsh"
+source "${VSI_COMMON_DIR}/linux/compat.bsh"
+source "${VSI_COMMON_DIR}/linux/dir_tools.bsh"
+
+#*# just/plugins/just_archive_functions
+
+JUST_DEFAULTIFY_FUNCTIONS+=(archive_defaultify)
+JUST_HELP_FILES+=("${BASH_SOURCE[0]}")
+
+#**
+# .. default-domain:: bash
+#
+# ==========================
+# J.U.S.T. Archive Functions
+# ==========================
+#
+# .. file:: just_archive_functions.bsh
+#
+# git archive plugin for just
+#**
+
+#**
+# .. envvar:: JUST_ARCHIVE_FILE
+#
+# Default archive file.
+#
+# .. seealso::
+#
+#   :func:`git-archive-recursive`
+#**
+
+: ${JUST_ARCHIVE_FILE=}
+
+#**
+# .. envvar:: JUST_ARCHIVE_DIR
+#
+# Default archive directory containing git repo.
+#
+# .. seealso::
+#
+#   :func:`git-archive-recursive`
+#**
+
+: ${JUST_ARCHIVE_DIR=}
+
+#**
+# .. function:: git-archive-recursive
+#
+# Recursive ``git archive`` on current repo including submodules, saving result to a tarball. The archive will be created from the current repo (i.e., ``HEAD``) & current submodules (i.e., ``HEAD`` of each submodule)
+#
+# :Arguments: * [``-o|--output {file}``] - Output archive file
+#             * [``--overwrite``] - Overwrite existing archive file
+#             * [``-C {path}``] - Run as if git was started in ``<path>`` instead of the current working directory.
+#
+# :Output: * archive_extra_args - number of arguments consumed
+#
+#**
+
+function git-archive-recursive()
+{
+  local archive_file="${JUST_ARCHIVE_FILE}"
+  local archive_dir="${JUST_ARCHIVE_DIR}"
+  local overwrite=0
+  local archive_extra_args
+  parse_args archive_extra_args \
+      --output archive_file: \
+      --overwrite overwrite \
+      -C archive_dir: \
+      -- ${@+"${@}"}
+
+  # check for overwrite
+  if [ -f "${archive_file}" ]; then
+    if [ "${overwrite}" != "0" ]; then
+      echo "WARNING: output archive file exists and will be overwritten " \
+           "<${archive_file}>" >&2
+    else
+      echo "ERROR: output archive file exists and overwrite is disabled "\
+           "<${archive_file}>" >&2
+      JUST_IGNORE_EXIT_CODES=1
+      return 1
+    fi
+  fi
+
+  # check that ``tar --concatenate`` is available
+  if ! tar_feature_concatenate; then
+    echo "ERROR: 'git-archive-recursive' requires 'tar --concatenate'" >&2
```

Comment:
Add something like "Considering installing a GNU version of tar and try again" to the message. Or do we need to mention setting the `TAR` variable to a GNU version of tar maybe?
```

Reply:
Expanded the comment